lib/tests/test_git.rs

```diff
@@ -1572,6 +1573,64 @@ fn test_export_reexport_transitions() {
     );
 }
 
+#[test]
+fn test_reset_head_to_root() {
```

Comment:
Are the CLI tests redundant now that we have this test?

Reply:
No. I haven't migrated the test for staging area, and I prefer to have basic e2e test because the implementation is cryptic.